src/wg/format/bmp.d

```diff
@@ -9,10 +9,51 @@ License:    $(WEB boost.org/LICENSE_1_0.txt, Boost License 1.0)
 
 module wg.format.bmp;
 
-import wg.image.imagebuffer;
+import wg.image;
 import wg.image.metadata;
+import wg.image.transform;
 import wg.util.allocator;
 
+/**
+Create an ImageBuffer from a BMP formatted image.
+*/
+ImageBuffer readBMP(void[] bmpBuffer)
+{
+    auto bmp = BMPImage(bmpBuffer);
+    ImageBuffer src = bmp.getImage();
+    return src.clone();
+}
+
+/**
+Create an ImageBuffer from a BMP formatted image.
+*/
+ImageBuffer readBMP(void[] bmpBuffer, Allocator* allocator) //nothrow @nogc
+{
+    auto bmp = BMPImage(bmpBuffer);
+    ImageBuffer src = bmp.getImage();
+    return src.clone(allocator);
```

Comment:
So bmp.getImage still uses GC and then it gets cloned with the given allocator?

Reply:
i'll rework the internal loader code. it certainly won't use GC, but yes, the thing you say is true right now. i *really* needed a source of test data and just hacked it together. needs a big tidy.
I'm mostly concerned with the user-facing API and making sure it feels okay.